tests/LogLevelTest.php

```
@@ -7,24 +7,85 @@
+        $this->assertNotSame(LogLevel::debug(), LogLevel::error());
+    }
+
+    public function testInstancesRejectMutation(): void
+    {
+        // Native enum cases are readonly; the polyfill must prevent writes too,
+        // otherwise mutating one call site would corrupt the cached singleton
+        // for every other caller.
+        $level = LogLevel::error();
+
+        $this->expectException(\LogicException::class);
+        $level->name = 'mutated';
```

PHPStan (2.1.51)

[error] Property Timewave\Logger\Enums\LogLevel::$name is not writable.


(assign.propertyReadOnly)